### PR TITLE
Change form fields on invite user form

### PR DESCRIPTION
**Pull request by Rosa-Fox**
- The invite user form should be displaying First / Last name over Given / Family name.
- Due to Cognito reasons we still should be using Family / Given name in the backend.

- Displays validations with updated content.
- Fixing displaying validations for the role checkboxes has also been snuck in...
- Linking from the error summary at the top of the form to the form fields was also previously broken and is now fixed.

invite-form

invite-form02

### CODE REVIEW

File: app/models/invite_user_form.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@ class InviteUserForm
 
   attr_reader :email, :given_name, :family_name, :roles
 
-  validates_presence_of :email, :given_name, :family_name, :roles
+  validates_presence_of :email, :roles
 
-  validate :email_is_valid, :validate_roles
+  validate :email_is_valid, :validate_roles, :first_name_is_present, :last_name_is_present
```
Comment (jakubmiarka) [inline app/models/invite_user_form.rb]: there's an easier way to customize the error message, you can still use the `validates_presence_of` method but you can pass in the message, e.g. `validates_presence_of :first_name, message: t('whatever.error_text')`.

Alternatively, we don't need to make any changes here and just use locales to overwrite the name of the field in locales like so:
```
en:
  activerecord:
    attributes:
      inviteuserform:
        given_name: "First name"
```

However, I think we could just rename everything to the new naming (first/last name) and only make sure on the API call to cognito it is given/family name...

Comment (Rosa-Fox) [reply app/models/invite_user_form.rb]: I wasn't sure if I should change it everywhere (apart from when we send to Cognito) or to just change it there on the view as `given_name` and `family_name` are used quite a lot throughout... will change now... and will localise the error text... should have done that before.

Comment (Rosa-Fox) [reply app/models/invite_user_form.rb]: @jakubmiarka I have amended the commit and changed the field names... `given_name` and `family_name` are still  being used by the `user` model and controller... I think that makes sense as they are fairly tied to cognito and also changing them and not cognito is proving difficult... though I can keep trying if you literally only want `given_name` to be sent and received from Cognito.

File: app/models/invite_user_form.rb [hunk 2]
```diff
@@ -1,16 +1,16 @@
 class InviteUserForm
   include ActiveModel::Model
 
-  attr_reader :email, :given_name, :family_name, :roles
+  attr_reader :email, :first_name, :last_name, :roles
 
-  validates_presence_of :email, :given_name, :family_name, :roles
+  validates_presence_of :email, :roles, :first_name, :last_name
```
Comment (jakubmiarka) [inline app/models/invite_user_form.rb]: can we keep the order of what's on the screen? just so when the error messages appear they're in the same order as the fields

File: app/models/invite_user_form.rb (hunk 1 above)
Comment (jakubmiarka) [reply app/models/invite_user_form.rb]: yeah, looks good! ... Cognito naming should not be tied to/affect our model namings...

File: app/models/invite_user_form.rb (hunk 2 above)
Comment (Rosa-Fox) [reply app/models/invite_user_form.rb]: thanks, have amended

File: app/models/invite_user_form.rb (hunk 1 above)
Comment (Rosa-Fox) [reply app/models/invite_user_form.rb]: @jakubmiarka I have updated it and changed the field naming in the user model. I think I have changed it as much as I can now. Let me know what you think :)

**Comment by AmritSidhu** [comment]
Chris also noticed that there is extra spacing between the label and the text box which i found is caused by this extra span, generated by the error message:
Screen Shot 2019-11-18 at 14 06 15

But as the error should only show when there is an error, not sure why its there. Is this something you can look into as part of this PR?

File: spec/models/invite_user_form_spec.rb [hunk 3]
```diff
@@ -3,26 +3,26 @@
 RSpec.describe InviteUserForm, type: :model do
 
   it 'is valid with valid attributes' do
-    expect(InviteUserForm.new({email: 'test@test.test', given_name: 'First Name', family_name: 'Surname', roles: [ROLE::CERTIFICATE_MANAGER] })).to be_valid
+    expect(InviteUserForm.new({email: 'test@test.test', first_name: 'First Name', last_name: 'Surname', roles: [ROLE::CERTIFICATE_MANAGER] })).to be_valid
```
Comment (jkunle) [inline spec/models/invite_user_form_spec.rb]: Nice job Rosa. Just very minor clean up
```
InviteUserForm.new({email: 'test@test.test', first_name: 'First Name', last_name: 'Surname', roles: [ROLE::CERTIFICATE_MANAGER] })```
```
to
```
InviteUserForm.new(email: 'test@test.test', first_name: 'First Name', last_name: 'Surname', roles: [ROLE::CERTIFICATE_MANAGER])
```

**Comment by Rosa-Fox** [comment]
> Chris also noticed that there is extra spacing between the label and the text box which i found is caused by this extra span, generated by the error message:
> Screen Shot 2019-11-18 at 14 06 15
> 
> But as the error should only show when there is an actual error, not sure why its there. Is this something you can look into as part of this PR?

@AmritSidhu good spot. I have fixed and amended.

**Pull request merged**